### PR TITLE
Don't ignore exceptions from tasks

### LINKED ISSUE #847: Nigthly run contains errors for Post correct credentials step
While looking into nightly run Gatling results from this run: https://github.com/keycloak/keycloak-benchmark/actions/runs/9493985920 I found out there is a small amount of errors for the Post correct credentials step.

The number of errors is small (140/91500), however, this is not expected. Keycloak should handle all requests. This is happening with all deployment flavours Active/Passive, Persistent sessions and Active/Active.

Here is a Gatling result for A/A:
image

The error response is the following:
```
<form id="kc-form-login" onsubmit="login.disabled = true; return true;" action="https://a7ec6e963d7b41329.awsglobalaccelerator.com/realms/realm-0/login-actions/authenticate?session_code=bum7MIojVl1oGn90P0LI4Z7aKSNkppsgw0un4giAjKs&amp;execution=c49cea85-d7b4-45b5-8fb6-171a85cf80fd&amp;client_id=client-0&amp;tab_id=ujaSzmqf9Xc&amp;client_data=eyJydSI6Imh0dHBzOi8vYTdlYzZlOTYzZDdiNDEzMjkuYXdzZ2xvYmFsYWNjZWxlcmF0b3IuY29tL3JlYWxtcy9yZWFsbS0wL2FjY291bnQiLCJydCI6ImNvZGUiLCJzdCI6ImViMzMzMDQzLWZhYzktM2MzZi0zMTA1LWJlOTY4MmRkOGM2NyJ9" method="post">
                        <div class="form-group">
                            <label for="username" class="pf-c-form__label pf-c-form__label-text">Username or email</label>

                            <input tabindex="2" id="username" class="pf-c-form-control" name="username" value="user-29"  type="text" autofocus autocomplete="username"
                                   aria-invalid="true"
                                   dir="ltr"
                            />

                                <span id="input-error" class="pf-c-form__helper-text pf-m-error required kc-feedback-text" aria-live="polite">
                                        Invalid username or password.
                                </span>

                        </div>

                    <div class="form-group">
                        <label for="password" class="pf-c-form__label pf-c-form__label-text">Password</label>

                        <div class="pf-c-input-group" dir="ltr">
                            <input tabindex="3" id="password" class="pf-c-form-control" name="password" type="password" autocomplete="current-password"
                                   aria-invalid="true"
                            />
                            <button class="pf-c-button pf-m-control" type="button" aria-label="Show password"
                                    aria-controls="password" data-password-toggle tabindex="4"
                                    data-icon-show="fa fa-eye" data-icon-hide="fa fa-eye-slash"
                                    data-label-show="Show password" data-label-hide="Hide password">
                                <i class="fa fa-eye" aria-hidden="true"></i>
                            </button>
                        </div>


                    </div>

                    <div class="form-group login-pf-settings">
                        <div id="kc-form-options">
                            </div>
                            <div class="">
                            </div>

                      </div>

                      <div id="kc-form-buttons" class="form-group">
                          <input type="hidden" id="id-hidden-input" name="credentialId" />
                          <input tabindex="7" class="pf-c-button pf-m-primary pf-m-block btn-lg" name="login" id="kc-login" type="submit" value="Sign In"/>
                      </div>
                </form>
```

Here is the full request response from Gatling log:
```
Request:
Browser posts correct credentials: KO status.find.is(302), but actually found 200
=========================
Session:
Session(Authentication - Authorization Code Username/Password,1954,HashMap(clientId -> client-0, username -> user-29, scope -> openid profile, adminUsername -> null, lastName -> , firstName -> , gatling.http.ssl.sslContexts -> io.gatling.http.util.SslContexts@4e24e0c0, adminPassword -> null, clientSecret -> client-0-secret, realm -> realm-0, keycloakServer -> https://a7ec6e963d7b41329.awsglobalaccelerator.com, login-form-uri -> https://a7ec6e963d7b41329.awsglobalaccelerator.com/realms/realm-0/login-actions/authenticate?session_code=iJG3dp25O0kcM9DakWVyDYW-oXL70qGfLkVgAAk5jv0&execution=c49cea85-d7b4-45b5-8fb6-171a85cf80fd&client_id=client-0&tab_id=ujaSzmqf9Xc&client_data=eyJydSI6Imh0dHBzOi8vYTdlYzZlOTYzZDdiNDEzMjkuYXdzZ2xvYmFsYWNjZWxlcmF0b3IuY29tL3JlYWxtcy9yZWFsbS0wL2FjY291bnQiLCJydCI6ImNvZGUiLCJzdCI6ImViMzMzMDQzLWZhYzktM2MzZi0zMTA1LWJlOTY4MmRkOGM2NyJ9, gatling.http.cookies -> CookieJar(Map(CookieKey(auth_session_id,a7ec6e963d7b41329.awsglobalaccelerator.com,/realms/realm-0/) -> StoredCookie(AUTH_SESSION_ID=32867078-21b5-4f45-a83e-dfc7947f511e.keycloak-0-39870, path=/realms/realm-0/, secure, HTTPOnly, SameSite=None,true,false,1718268114709), CookieKey(auth_session_id_legacy,a7ec6e963d7b41329.awsglobalaccelerator.com,/realms/realm-0/) -> StoredCookie(AUTH_SESSION_ID_LEGACY=32867078-21b5-4f45-a83e-dfc7947f511e.keycloak-0-39870, path=/realms/realm-0/, secure, HTTPOnly,true,false,1718268114709), CookieKey(kc_restart,a7ec6e963d7b41329.awsglobalaccelerator.com,/realms/realm-0/) -> StoredCookie(KC_RESTART=eyJhbGciOiJkaXIiLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0..H4OTehssb9tjtBWVpSb_7w.tYoycOwGm5cacqTZVWEhCLU12sVZsk4uGGB1MNSe7Mhl4JaURoPd5iDk6Bd-NJ6IxMvv07W1w0rc1GeZTgQtWm_pjpxslA7vevhKytG3_KcWtsSRvh_XgVJDomp6uWx5dQs7vEWQlTHglmgdUWwHUwvBmXnt5rjVzrnc6clmmal-aVMsr02zXPDU-MG6sB0rrL-WG29wfkdVkiu-sxdEyLOiYtagmN62mM6qQgUnbKv-pglNP9gJPbqVV-mWNQAbNcgi_PWskNcLpAyh4mUMKPSIl83LDmDh8VwszGlDSp4jP-k0TkJreWETdHepzx0vdHwTKF5_uMgmqfMBpc8sBgXRSF1_QbGXXsUZWGC-9a9MlyMuFfea-7h8zcWg9X7Xs2SElkUi0QsxLY9yPF1OvDFqjESPUe_wREIKZ9nMViq5gKSwHA2GBsQU5CqqGvws6s4_G24EaOz1YV8LXzJ7datkZDuQpCUzhRVMO1hA8MBX8ttRTEllVoyi1Mk8Pd1PRzy-gOeFX29Uq-umvS4pKuG9dQ_UPrLOVJD1QlkJ0dxepshaZcX_idF4d7PjulU11B_Q1xXK-ALge0SFi3GenIbmKivn3kLQYPahgBLKWR1gjBNsnKeAob6JYXWJk4bNA94vmpFu6uOz7i39epmCE1DtegAgc3DKj4HmiJX--GN-ZZfhcl5Jami2rMT0MJ5YKfIExB-LmRmiPxWNj7RgRr_eONL4uYyE0N3dA07229bktuA5Lj_m9tPsNJWJTxWfavkOKhyjclOWHlp8wjcQgbMKUYmvc99Yse6splftdxyQ54abxz_eM2el5VV-ReSjPoSFWe5yJOesPW_jLs6PcxFX6V_t1OI0uEBIEijWP4qAjhQLDwmYE8pJMMyUD6qEqWqnF8ztI-gBy-XrGnC4W7eFKrP2XTbyQAG9DX9HsH31TQYY33QNjXPyhS3CRYqhpdTF23JM6JKbrktsMP3BiQP2KRINPcQTbAFMJHGQlaCRPPAOV1rxXrmkR6a8Crj70jzhIZCi3SXjprK2RKzeelNzeXv3prdcuiyZ8YzIN5U.xckOl09OLgD0zdLI_Pg0Fw, path=/realms/realm-0/, secure, HTTPOnly, SameSite=None,true,false,1718268114709))), password -> user-29-password, email -> , gatling.http.cache.dns -> io.gatling.http.resolver.ShufflingNameResolver@3326ca7f, state -> eb333043-fac9-3c3f-3105-be9682dd8c67, redirectUri -> https://a7ec6e963d7b41329.awsglobalaccelerator.com/realms/realm-0/account, wrongPasswordCount -> 0, gatling.http.referer -> https://a7ec6e963d7b41329.awsglobalaccelerator.com/realms/realm-0/login-actions/authenticate?session_code=iJG3dp25O0kcM9DakWVyDYW-oXL70qGfLkVgAAk5jv0&execution=c49cea85-d7b4-45b5-8fb6-171a85cf80fd&client_id=client-0&tab_id=ujaSzmqf9Xc&client_data=eyJydSI6Imh0dHBzOi8vYTdlYzZlOTYzZDdiNDEzMjkuYXdzZ2xvYmFsYWNjZWxlcmF0b3IuY29tL3JlYWxtcy9yZWFsbS0wL2FjY291bnQiLCJydCI6ImNvZGUiLCJzdCI6ImViMzMzMDQzLWZhYzktM2MzZi0zMTA1LWJlOTY4MmRkOGM2NyJ9),KO,List(),io.gatling.core.protocol.ProtocolComponentsRegistry$$Lambda$751/0x0000000800649108@75137016,io.netty.channel.epoll.EpollEventLoop@59546cfe)
=========================
HTTP request:
POST https://a7ec6e963d7b41329.awsglobalaccelerator.com/realms/realm-0/login-actions/authenticate?session_code=iJG3dp25O0kcM9DakWVyDYW-oXL70qGfLkVgAAk5jv0&execution=c49cea85-d7b4-45b5-8fb6-171a85cf80fd&client_id=client-0&tab_id=ujaSzmqf9Xc&client_data=eyJydSI6Imh0dHBzOi8vYTdlYzZlOTYzZDdiNDEzMjkuYXdzZ2xvYmFsYWNjZWxlcmF0b3IuY29tL3JlYWxtcy9yZWFsbS0wL2FjY291bnQiLCJydCI6ImNvZGUiLCJzdCI6ImViMzMzMDQzLWZhYzktM2MzZi0zMTA1LWJlOTY4MmRkOGM2NyJ9
headers:
	accept: text/html,application/xhtml+xml,application/xml
	Accept-Encoding: gzip, deflate
	Accept-Language: en-US,en;q=0.5
	User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.8; rv:16.0) Gecko/20100101 Firefox/16.0
	referer: https://a7ec6e963d7b41329.awsglobalaccelerator.com/realms/realm-0/protocol/openid-connect/auth?login=true&response_type=code&client_id=client-0&state=eb333043-fac9-3c3f-3105-be9682dd8c67&redirect_uri=https%3A%2F%2Fa7ec6e963d7b41329.awsglobalaccelerator.com%2Frealms%2Frealm-0%2Faccount&scope=openid+profile
	cookie: AUTH_SESSION_ID=32867078-21b5-4f45-a83e-dfc7947f511e.keycloak-0-39870; AUTH_SESSION_ID_LEGACY=32867078-21b5-4f45-a83e-dfc7947f511e.keycloak-0-39870; KC_RESTART=eyJhbGciOiJkaXIiLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0..H4OTehssb9tjtBWVpSb_7w.tYoycOwGm5cacqTZVWEhCLU12sVZsk4uGGB1MNSe7Mhl4JaURoPd5iDk6Bd-NJ6IxMvv07W1w0rc1GeZTgQtWm_pjpxslA7vevhKytG3_KcWtsSRvh_XgVJDomp6uWx5dQs7vEWQlTHglmgdUWwHUwvBmXnt5rjVzrnc6clmmal-aVMsr02zXPDU-MG6sB0rrL-WG29wfkdVkiu-sxdEyLOiYtagmN62mM6qQgUnbKv-pglNP9gJPbqVV-mWNQAbNcgi_PWskNcLpAyh4mUMKPSIl83LDmDh8VwszGlDSp4jP-k0TkJreWETdHepzx0vdHwTKF5_uMgmqfMBpc8sBgXRSF1_QbGXXsUZWGC-9a9MlyMuFfea-7h8zcWg9X7Xs2SElkUi0QsxLY9yPF1OvDFqjESPUe_wREIKZ9nMViq5gKSwHA2GBsQU5CqqGvws6s4_G24EaOz1YV8LXzJ7datkZDuQpCUzhRVMO1hA8MBX8ttRTEllVoyi1Mk8Pd1PRzy-gOeFX29Uq-umvS4pKuG9dQ_UPrLOVJD1QlkJ0dxepshaZcX_idF4d7PjulU11B_Q1xXK-ALge0SFi3GenIbmKivn3kLQYPahgBLKWR1gjBNsnKeAob6JYXWJk4bNA94vmpFu6uOz7i39epmCE1DtegAgc3DKj4HmiJX--GN-ZZfhcl5Jami2rMT0MJ5YKfIExB-LmRmiPxWNj7RgRr_eONL4uYyE0N3dA07229bktuA5Lj_m9tPsNJWJTxWfavkOKhyjclOWHlp8wjcQgbMKUYmvc99Yse6splftdxyQ54abxz_eM2el5VV-ReSjPoSFWe5yJOesPW_jLs6PcxFX6V_t1OI0uEBIEijWP4qAjhQLDwmYE8pJMMyUD6qEqWqnF8ztI-gBy-XrGnC4W7eFKrP2XTbyQAG9DX9HsH31TQYY33QNjXPyhS3CRYqhpdTF23JM6JKbrktsMP3BiQP2KRINPcQTbAFMJHGQlaCRPPAOV1rxXrmkR6a8Crj70jzhIZCi3SXjprK2RKzeelNzeXv3prdcuiyZ8YzIN5U.xckOl09OLgD0zdLI_Pg0Fw
	origin: https://a7ec6e963d7b41329.awsglobalaccelerator.com
	host: a7ec6e963d7b41329.awsglobalaccelerator.com
	content-type: application/x-www-form-urlencoded
	content-length: 55
cookies:
	AUTH_SESSION_ID=32867078-21b5-4f45-a83e-dfc7947f511e.keycloak-0-39870, path=/realms/realm-0/, secure, HTTPOnly, SameSite=None
	AUTH_SESSION_ID_LEGACY=32867078-21b5-4f45-a83e-dfc7947f511e.keycloak-0-39870, path=/realms/realm-0/, secure, HTTPOnly
	KC_RESTART=eyJhbGciOiJkaXIiLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0..H4OTehssb9tjtBWVpSb_7w.tYoycOwGm5cacqTZVWEhCLU12sVZsk4uGGB1MNSe7Mhl4JaURoPd5iDk6Bd-NJ6IxMvv07W1w0rc1GeZTgQtWm_pjpxslA7vevhKytG3_KcWtsSRvh_XgVJDomp6uWx5dQs7vEWQlTHglmgdUWwHUwvBmXnt5rjVzrnc6clmmal-aVMsr02zXPDU-MG6sB0rrL-WG29wfkdVkiu-sxdEyLOiYtagmN62mM6qQgUnbKv-pglNP9gJPbqVV-mWNQAbNcgi_PWskNcLpAyh4mUMKPSIl83LDmDh8VwszGlDSp4jP-k0TkJreWETdHepzx0vdHwTKF5_uMgmqfMBpc8sBgXRSF1_QbGXXsUZWGC-9a9MlyMuFfea-7h8zcWg9X7Xs2SElkUi0QsxLY9yPF1OvDFqjESPUe_wREIKZ9nMViq5gKSwHA2GBsQU5CqqGvws6s4_G24EaOz1YV8LXzJ7datkZDuQpCUzhRVMO1hA8MBX8ttRTEllVoyi1Mk8Pd1PRzy-gOeFX29Uq-umvS4pKuG9dQ_UPrLOVJD1QlkJ0dxepshaZcX_idF4d7PjulU11B_Q1xXK-ALge0SFi3GenIbmKivn3kLQYPahgBLKWR1gjBNsnKeAob6JYXWJk4bNA94vmpFu6uOz7i39epmCE1DtegAgc3DKj4HmiJX--GN-ZZfhcl5Jami2rMT0MJ5YKfIExB-LmRmiPxWNj7RgRr_eONL4uYyE0N3dA07229bktuA5Lj_m9tPsNJWJTxWfavkOKhyjclOWHlp8wjcQgbMKUYmvc99Yse6splftdxyQ54abxz_eM2el5VV-ReSjPoSFWe5yJOesPW_jLs6PcxFX6V_t1OI0uEBIEijWP4qAjhQLDwmYE8pJMMyUD6qEqWqnF8ztI-gBy-XrGnC4W7eFKrP2XTbyQAG9DX9HsH31TQYY33QNjXPyhS3CRYqhpdTF23JM6JKbrktsMP3BiQP2KRINPcQTbAFMJHGQlaCRPPAOV1rxXrmkR6a8Crj70jzhIZCi3SXjprK2RKzeelNzeXv3prdcuiyZ8YzIN5U.xckOl09OLgD0zdLI_Pg0Fw, path=/realms/realm-0/, secure, HTTPOnly, SameSite=None
body:FormUrlEncodedRequestBody{patchedContentType='application/x-www-form-urlencoded', charset=UTF-8, content=username=user-29&password=user-29-password&login=Log+in}
=========================
HTTP response:
version:
	HTTP/1.1
status:
	200 OK
headers:
	Cache-Control: no-store, must-revalidate, max-age=0
	content-length: 6648
	Content-Language: en
	Content-Security-Policy: frame-src 'self'; frame-ancestors 'self'; object-src 'none';
	Content-Type: text/html;charset=utf-8
	Referrer-Policy: no-referrer
	Strict-Transport-Security: max-age=31536000; includeSubDomains
	X-Content-Type-Options: nosniff
	X-Frame-Options: SAMEORIGIN
	X-Robots-Tag: none
	X-XSS-Protection: 1; mode=block

body:
<!DOCTYPE html>
<html class="login-pf">

<head>
    <meta charset="utf-8">
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
    <meta name="robots" content="noindex, nofollow">

            <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <title>Sign in to realm-0</title>
    <link rel="icon" href="/resources/1r9a8/login/keycloak/img/favicon.ico" />
            <link href="/resources/1r9a8/common/keycloak/node_modules/@patternfly/patternfly/patternfly.min.css" rel="stylesheet" />
            <link href="/resources/1r9a8/common/keycloak/node_modules/patternfly/dist/css/patternfly.min.css" rel="stylesheet" />
            <link href="/resources/1r9a8/common/keycloak/node_modules/patternfly/dist/css/patternfly-additions.min.css" rel="stylesheet" />
            <link href="/resources/1r9a8/common/keycloak/lib/pficon/pficon.css" rel="stylesheet" />
            <link href="/resources/1r9a8/login/keycloak/css/login.css" rel="stylesheet" />
    <script type="importmap">
        {
            "imports": {
                "rfc4648": "/resources/1r9a8/common/keycloak/node_modules/rfc4648/lib/rfc4648.js"
            }
        }
    </script>
    <script src="/resources/1r9a8/login/keycloak/js/menu-button-links.js" type="module"></script>
    <script type="module">
        import { checkCookiesAndSetTimer } from "/resources/1r9a8/login/keycloak/js/authChecker.js";

        checkCookiesAndSetTimer(
          "/realms/realm-0/login-actions/restart?client_id=client-0&tab_id=ujaSzmqf9Xc&client_data=eyJydSI6Imh0dHBzOi8vYTdlYzZlOTYzZDdiNDEzMjkuYXdzZ2xvYmFsYWNjZWxlcmF0b3IuY29tL3JlYWxtcy9yZWFsbS0wL2FjY291bnQiLCJydCI6ImNvZGUiLCJzdCI6ImViMzMzMDQzLWZhYzktM2MzZi0zMTA1LWJlOTY4MmRkOGM2NyJ9&skip_logout=true"
        );
    </script>
<SCRIPT> if (typeof history.replaceState === 'function') {  history.replaceState({}, "some title", "https://a7ec6e963d7b41329.awsglobalaccelerator.com/realms/realm-0/login-actions/authenticate?execution=c49cea85-d7b4-45b5-8fb6-171a85cf80fd&client_id=client-0&tab_id=ujaSzmqf9Xc&client_data=eyJydSI6Imh0dHBzOi8vYTdlYzZlOTYzZDdiNDEzMjkuYXdzZ2xvYmFsYWNjZWxlcmF0b3IuY29tL3JlYWxtcy9yZWFsbS0wL2FjY291bnQiLCJydCI6ImNvZGUiLCJzdCI6ImViMzMzMDQzLWZhYzktM2MzZi0zMTA1LWJlOTY4MmRkOGM2NyJ9"); }</SCRIPT></head>

<body class="">
<div class="login-pf-page">
    <div id="kc-header" class="login-pf-page-header">
        <div id="kc-header-wrapper"
             class="">realm-0</div>
    </div>
    <div class="card-pf">
        <header class="login-pf-header">
                <h1 id="kc-page-title">        Sign in to your account

</h1>
      </header>
      <div id="kc-content">
        <div id="kc-content-wrapper">


        <div id="kc-form">
          <div id="kc-form-wrapper">
                <form id="kc-form-login" onsubmit="login.disabled = true; return true;" action="https://a7ec6e963d7b41329.awsglobalaccelerator.com/realms/realm-0/login-actions/authenticate?session_code=bum7MIojVl1oGn90P0LI4Z7aKSNkppsgw0un4giAjKs&amp;execution=c49cea85-d7b4-45b5-8fb6-171a85cf80fd&amp;client_id=client-0&amp;tab_id=ujaSzmqf9Xc&amp;client_data=eyJydSI6Imh0dHBzOi8vYTdlYzZlOTYzZDdiNDEzMjkuYXdzZ2xvYmFsYWNjZWxlcmF0b3IuY29tL3JlYWxtcy9yZWFsbS0wL2FjY291bnQiLCJydCI6ImNvZGUiLCJzdCI6ImViMzMzMDQzLWZhYzktM2MzZi0zMTA1LWJlOTY4MmRkOGM2NyJ9" method="post">
                        <div class="form-group">
                            <label for="username" class="pf-c-form__label pf-c-form__label-text">Username or email</label>

                            <input tabindex="2" id="username" class="pf-c-form-control" name="username" value="user-29"  type="text" autofocus autocomplete="username"
                                   aria-invalid="true"
                                   dir="ltr"
                            />

                                <span id="input-error" class="pf-c-form__helper-text pf-m-error required kc-feedback-text" aria-live="polite">
                                        Invalid username or password.
                                </span>

                        </div>

                    <div class="form-group">
                        <label for="password" class="pf-c-form__label pf-c-form__label-text">Password</label>

                        <div class="pf-c-input-group" dir="ltr">
                            <input tabindex="3" id="password" class="pf-c-form-control" name="password" type="password" autocomplete="current-password"
                                   aria-invalid="true"
                            />
                            <button class="pf-c-button pf-m-control" type="button" aria-label="Show password"
                                    aria-controls="password" data-password-toggle tabindex="4"
                                    data-icon-show="fa fa-eye" data-icon-hide="fa fa-eye-slash"
                                    data-label-show="Show password" data-label-hide="Hide password">
                                <i class="fa fa-eye" aria-hidden="true"></i>
                            </button>
                        </div>


                    </div>

                    <div class="form-group login-pf-settings">
                        <div id="kc-form-options">
                            </div>
                            <div class="">
                            </div>

                      </div>

                      <div id="kc-form-buttons" class="form-group">
                          <input type="hidden" id="id-hidden-input" name="credentialId" />
                          <input tabindex="7" class="pf-c-button pf-m-primary pf-m-block btn-lg" name="login" id="kc-login" type="submit" value="Sign In"/>
                      </div>
                </form>
            </div>
        </div>
        <script type="module" src="/resources/1r9a8/login/keycloak/js/passwordVisibility.js"></script>





              <div id="kc-info" class="login-pf-signup">
                  <div id="kc-info-wrapper" class="">
            <div id="kc-registration-container">
                <div id="kc-registration">
                    <span>New user? <a tabindex="8"
                                                 href="/realms/realm-0/login-actions/registration?client_id=client-0&amp;tab_id=ujaSzmqf9Xc&amp;client_data=eyJydSI6Imh0dHBzOi8vYTdlYzZlOTYzZDdiNDEzMjkuYXdzZ2xvYmFsYWNjZWxlcmF0b3IuY29tL3JlYWxtcy9yZWFsbS0wL2FjY291bnQiLCJydCI6ImNvZGUiLCJzdCI6ImViMzMzMDQzLWZhYzktM2MzZi0zMTA1LWJlOTY4MmRkOGM2NyJ9">Register</a></span>
                </div>
            </div>

                  </div>
              </div>
        </div>
      </div>

    </div>
  </div>
</body>
</html>
```

### PR DESCRIPTION
**Pull request by ahus1**
According the the JavaDoc a future is also done when it failed. The current code will only show exceptions if a task that the code is waiting for is failing with an exception. Any other task that failed in the meantime is silently ignored.

Closes #847

### CODE REVIEW

**Comment by ahus1** [comment]
@kami619 - you've been involved in the original issue, maybe you could review this PR?

File: dataset/src/main/java/org/keycloak/benchmark/dataset/ExecutorHelper.java [hunk 1]
```diff
@@ -50,29 +50,35 @@ public ExecutorHelper(int threadCount, KeycloakSessionFactory sessionFactory, Da
...
+        while (!futures.isEmpty()) {
+            // some tasks will spawn other tasks, so expect the futures queue to be modified concurrently.
+            try {
+                futures.poll().get();
+            } catch (Throwable e) {
+                failures.add(e);
+            }
+        }
+        if (!failures.isEmpty()) {
+            RuntimeException ex = new RuntimeException("Some futures failed");
```
Comment (kami619) [inline dataset/src/main/java/org/keycloak/benchmark/dataset/ExecutorHelper.java:75]: would it be possible to log more information about the type of failure which could help with the debug or would it log anyways ?

Comment (ahus1) [reply dataset/src/main/java/org/keycloak/benchmark/dataset/ExecutorHelper.java:75]: The code creates a Runtime Exception which then has a list of one or more suppressed exceptions. All of the suppressed exceptions will be logged to together with the message "Some futures failed".

The the following line for where the suppressed exceptions are added.

**Review by mhajas — APPROVED** [review]
Thanks for the change. Let's see what is happening there tomorrow.

**Pull request merged**